src/contents.ts

```diff
@@ -353,6 +379,42 @@ class ContentsManager implements IContents.IManager {
   }
 
   /**
+   * Get a path to a file given a relative path.
+   *
+   * @param relativePath - The relative path to the file.
+   *
+   * @param basePath - The optional base path of the file.  The
+   *   default is the server root path.
+   *
+   * #### Notes
+   * If the path is not contained within the server path, it will be
+   * returned unmodified.
+   * If the relativePath starts with a forward slash, the basePath will
+   * be ignored and a path relative to the server root will be returned.
+   */
+  getPath(relativePath: string, basePath = ''): string {
+    let norm = pathutil.normalize(pathutil.join(basePath, relativePath));
+    if (norm.indexOf('..' + pathutil.sep) === 0) {
```

Comment:
Yeah, I had the hair-brained idea to make this "cross platform" last night, fixing now ;).